web/src/containers/QuickExchange/index.tsx

```diff
@@ -0,0 +1,373 @@
+import React, { useCallback, useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { useIntl } from 'react-intl';
+import { Button } from 'react-bootstrap';
+import { msPricesUpdates } from '../../api';
+import { useCurrenciesFetch } from '../../hooks';
+import {
+    marketsFetch,
+    selectMarkets,
+    selectWallets,
+    walletsFetch,
+    marketPriceFetch,
+    selectMarketPrice,
+    selectMarketPriceFetchSuccess,
+    createQuickExchangeFetch,
+} from '../../modules'
+import { SwipeIcon } from '../../assets/images/swipe';
+import { Decimal, CurrencyIcon, QuickExchangeForm, DropdownComponent, Timer } from '../../components';
+import {
+    getCurrencyForMarket,
+    getCurrencyFiltred,
+    getMarket,
+    getWallets,
+    getWallet,
+    getBaseAmount,
+    getQuoteAmount,
+} from './helpers';
+import { precisionRegExp, ssToMMSS } from '../../helpers';
+import { ArrowRight, WarningIcon } from './icons';
+
+interface QuickExchangeTimer {
+    initialSeconds: number;
+    initialMinutes: number;
+};
+
+interface QuickExchangeInterface {
+    amount: string;
+    currency: string;
+};
+
+const DEFAULT_VALUE_TIMER: QuickExchangeTimer = {
+    initialSeconds: 0,
+    initialMinutes: 0,
+};
+
+const DEFAULT_VALUE: QuickExchangeInterface = {
+    amount: '',
+    currency: '',
+};
+
+export const QuickExchangeContainer = () => {
+    const [base, setBaseData] = useState<QuickExchangeInterface>(DEFAULT_VALUE);
+    const [quote, setQuoteData] = useState<QuickExchangeInterface>(DEFAULT_VALUE);
+    const [type, setType] = useState('buy');
+    const [dropdownKey, setDropdownKey] = useState('');
+    const [marketID, setMarket] = useState('');
+    const [basePrecision, setBasePrecision] = useState(0);
+    const [quotePrecision, setQuotePrecision] = useState(0);
+    const [time, setTime] = useState<QuickExchangeTimer>(DEFAULT_VALUE_TIMER);
```

Comment:
This is ok, but better to use `useReducer` instead of `useState` for such case when we have more than 2 state variables.